# Part 3: NLP Pipeline & Sequence Modeling

## Task 1: Dataset Understanding

In [1]:
import pandas as pd

df = pd.read_csv('Data_Sets/part_3_nlp_sequence_modeling/customer_support_text_classification.csv')
print(f"Dataset shape: {df.shape}")
print(f"Number of records: {len(df)}")
print(f"\nColumns: {df.columns.tolist()}")
print(f"\nData types:\n{df.dtypes}")
print(f"\nMissing values:\n{df.isnull().sum()}")

Dataset shape: (1500, 6)
Number of records: 1500

Columns: ['ticket_id', 'channel', 'customer_message', 'sentiment_label', 'word_count', 'urgent_flag']

Data types:
ticket_id             str
channel               str
customer_message      str
sentiment_label       str
word_count          int64
urgent_flag         int64
dtype: object

Missing values:
ticket_id           0
channel             0
customer_message    0
sentiment_label     0
word_count          0
urgent_flag         0
dtype: int64


In [3]:
# Sample text records

print("Sample text records:")
for label in df['sentiment_label'].unique():
    sample = df[df['sentiment_label'] == label]['customer_message'].iloc[0]
    print(f"\n[{label.upper()}]: {sample}")

Sample text records:

[NEUTRAL]: I need information about the payment process. My ticket number is 78732. Please respond as soon as possible.

[POSITIVE]: The refund process was fast and convenient. I appreciate the quick response.

[NEGATIVE]: My refund is still pending and this experience is frustrating. My ticket number is 33927.


In [5]:
# Target labels/classes and Class distribution

print("Target labels/classes:", df['sentiment_label'].unique())
print(f"Number of classes: {df['sentiment_label'].nunique()}")
print(f"\nClass distribution:")
print(df['sentiment_label'].value_counts())
print(f"\nClass distribution (%):")
print(df['sentiment_label'].value_counts(normalize=True).round(3) * 100)

Target labels/classes: <StringArray>
['neutral', 'positive', 'negative']
Length: 3, dtype: str
Number of classes: 3

Class distribution:
sentiment_label
neutral     524
negative    497
positive    479
Name: count, dtype: int64

Class distribution (%):
sentiment_label
neutral     34.9
negative    33.1
positive    31.9
Name: proportion, dtype: float64


**Observations:**
1. The dataset has 1500 records with 3 sentiment classes.
2. Classes are fairly balanced (neutral: ~35%, negative: ~33%, positive: ~32%).
3. There are no missing values in the dataset
4. Messages are relatively short customer support texts.

## Task 2: Text Preprocessing